src/OrchardCore/Orchard.StorageProviders.FileSystem/FileSystemFile.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.IO;
+
+namespace Orchard.StorageProviders.FileSystem
+{
+    public class FileSystemFile : IFile
+    {
+        private readonly FileSystemInfo _fileInfo;
+        private readonly string _path;
+        private readonly string _publicPathPrefix;
+
+        public FileSystemFile(string path, string publicPathPrefix, FileSystemInfo fileInfo)
+        {
+            _fileInfo = fileInfo ?? throw new ArgumentNullException(nameof(fileInfo));
+            _path = path.Replace("\\", "/");
+            _publicPathPrefix = publicPathPrefix.Replace("\\", "/");
+        }
+
+        public string AbsolutePath => _publicPathPrefix + "/" + _path;
```

Comment:
Same answer, and Path.Combine won't just concatenate, if the second argument starts with '/', it will resolve it as the root.